app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.test.ts

```diff
@@ -0,0 +1,191 @@
+        dataDeletionService: {
+          createDataDeletionRegulationTask: jest
+            .fn()
+            .mockResolvedValue(mockTaskId),
+          fetchDeletionRegulationStatus: jest.fn().mockResolvedValue('UNKNOWN'),
+        },
+      });
+
+      await controller.createMetaMetricsDataDeletionTask();
+      expect(
+        dataDeletionService.createDataDeletionRegulationTask,
+      ).toHaveBeenCalledWith(mockMetaMetricsId);
+      expect(
+        dataDeletionService.createDataDeletionRegulationTask,
+      ).toHaveBeenCalledTimes(1);
+      expect(
+        dataDeletionService.fetchDeletionRegulationStatus,
+      ).toHaveBeenCalledTimes(1);
+      expect(controller.state).toMatchObject({
+        metaMetricsDataDeletionId: mockTaskId,
+        metaMetricsDataDeletionDate: expect.any(Number),
+        metaMetricsDataDeletionStatus: 'UNKNOWN',
+        participateInMetricsDuringDeletion: true,
+      });
+    });
+    it('creates a data deletion task and stores ID when user is not currently participating in metrics tracking', async () => {
+      const mockMetaMetricsId = 'mockId';
+      const mockTaskId = 'mockTaskId';
+      const { controller, dataDeletionService } = setupController({
+        options: {
+          getMetaMetricsId: jest.fn().mockReturnValue(mockMetaMetricsId),
+          getParticipateInMetrics: jest.fn().mockReturnValue(false),
+        },
+        dataDeletionService: {
+          createDataDeletionRegulationTask: jest
+            .fn()
+            .mockResolvedValue(mockTaskId),
+          fetchDeletionRegulationStatus: jest.fn().mockResolvedValue('UNKNOWN'),
+        },
+      });
+
+      await controller.createMetaMetricsDataDeletionTask();
+
+      expect(
+        dataDeletionService.createDataDeletionRegulationTask,
+      ).toHaveBeenCalledTimes(1);
+      expect(
+        dataDeletionService.fetchDeletionRegulationStatus,
+      ).toHaveBeenCalledTimes(1);
+      expect(
+        dataDeletionService.createDataDeletionRegulationTask,
+      ).toHaveBeenCalledWith(mockMetaMetricsId);
+      expect(controller.state).toMatchObject({
+        metaMetricsDataDeletionId: mockTaskId,
+        metaMetricsDataDeletionDate: expect.any(Number),
+        metaMetricsDataDeletionStatus: 'UNKNOWN',
+        participateInMetricsDuringDeletion: false,
+      });
+    });
+
+    it('fails to creates a data deletion task when user has never participating in metrics tracking', async () => {
+      const { controller } = setupController({
+        options: {
+          getMetaMetricsId: jest.fn().mockReturnValue(null),
+          getParticipateInMetrics: jest.fn().mockReturnValue(null),
+        },
+      });
+      await expect(
+        controller.createMetaMetricsDataDeletionTask(),
+      ).rejects.toThrow();
+      expect(controller.state).toMatchObject({
+        metaMetricsDataDeletionId: null,
+        metaMetricsDataDeletionDate: expect.any(Number),
+        participateInMetricsDuringDeletion: null,
+      });
```

Comment:
Are we sure we want to use `toMatchObject` instead of `toStrictEqual` here? This test will pass even if `controller.state` includes the `metaMetricsDataDeletionStatus` property.

I think it would be safer to use `toStrictEqual` throughout this test file as well, to ensure that we don't get passing tests in the presence of superfluous state properties.

```suggestion
      expect(controller.state).toStrictEqual({
        metaMetricsDataDeletionId: null,
        metaMetricsDataDeletionDate: expect.any(Number),
        participateInMetricsDuringDeletion: null,
      });
```